services/analysis-cached.js

```diff
@@ -305,99 +313,123 @@ const getLocationSelect = ({
-  if (paramKeysFiltered && paramKeysFiltered.length) {
-    let paramString = 'WHERE ';
-    paramKeysFiltered.forEach((p, i) => {
-      const isLast = paramKeysFiltered.length - 1 === i;
-      const isPolyname = ['forestType', 'landCategory'].includes(p);
-      const value = isPolyname ? 1 : params[p];
-      const polynameMeta = allPolynames.find(
-        (pname) => pname.value === params[p]
-      );
-      const tableKey =
-        polynameMeta &&
-        (polynameMeta.tableKey || polynameMeta.tableKeys[dataset || 'annual']);
-
-      /* TODO
-       perform better casting / allow to configure types:
-       AS for example wdpa_protected_area__id needs to be a string,
-       even that it evaluates AS a number.
-       Note that the postgres tables will allow us to cast at the query level.
-      */
-      // const zeroString = polynameMeta?.dataType === 'keyword' ? "'0'" : '0';
-      let isNumericValue = !!(
-        typeof value === 'number' ||
-        (!isNaN(value) && !['adm0', 'confidence'].includes(p))
-      );
-
-      let paramKey = p;
-      if (p === 'confidence') paramKey = 'confidence__cat';
-      if (p === 'threshold') {
-        //   paramKey = 'umd_tree_cover_density__threshold';
-        comparisonString = ' = ';
-
-        if (dataset === 'tropicalTreeCover') {
-          paramKey = 'wri_tropical_tree_cover__decile';
-        } else {
-          paramKey = 'umd_tree_cover_density_2000__threshold';
-        }
-
-        // }
-      }
-      if (p === 'adm0' && type === 'country') paramKey = 'iso';
-      if (p === 'adm1' && type === 'country') paramKey = 'adm1';
-      if (p === 'adm2' && type === 'country') paramKey = 'adm2';
-      if (p === 'adm0' && type === 'geostore') paramKey = 'geostore__id';
-      if (p === 'adm0' && type === 'wdpa') {
-        paramKey = 'wdpa_protected_area__id';
-        isNumericValue = false;
+
+  const allFilterOptions = forestTypes.concat(landCategories);
+  const allowedParams = ALLOWED_PARAMS[params.dataset || 'annual'];
+  const isTreeCoverDensity = dataset === 'treeCoverDensity';
+  const comparisonString = ' = ';
+
+  let paramString = 'WHERE ';
+  let paramKeys = Object.keys(params).filter((parameterName) => {
+    return (
+      (params[parameterName] || parameterName === 'threshold') &&
+      allowedParams.includes(parameterName)
+    );
+  });
+
+  if (!paramKeys?.length) {
+    return '';
+  }
+
+  /*
+   * Removing threshold from Tree Cover Density request
+   * Tree Cover Density has a default threshold >=10
+   */
+  if (isTreeCoverDensity && paramKeys.includes('threshold')) {
+    paramKeys = paramKeys.filter((item) => item !== 'threshold');
+  }
+
+  paramKeys.forEach((parameter, index) => {
+    const isLastParameter = paramKeys.length - 1 === index;
+    const hasFilterOption = ['forestType', 'landCategory'].includes(parameter);
+    const value = hasFilterOption ? 1 : params[parameter];
+    const polynameMeta = allFilterOptions.find(
```

Comment:
so, we still don't know what `polyName` refers to 🤷‍♂️

Reply:
I removed "polyname" variables but not all of them. Polynames for this method were filterOptions from `landCategory` or `forestType`. A weird name for a weird variable. We can remove all references of it later.